core/server/api/canary/posts.js

```diff
@@ -149,9 +149,18 @@ module.exports = {
                         return model;
                     }
 
-                    if (!model.get('email') && (model.get('status') === 'published') && model.wasChanged()) {
-                        const email = await mega.addEmail(model.toJSON());
-                        model.set('email', email);
+                    const postPublished = model.wasChanged() && (model.get('status') === 'published') && (model.previous('status') !== 'published');
```

Comment:
The controller logic starts to bloat a bit with email handling stuff. Would need to think up on how to handle it better if more code is added.